Feature/subscriptions (#875)
mercadopago/devsite-docs: guides/online-payments/subscriptions/integration.pt.md

`auto_recurring.frequency_type` (obrigatório) | Indica o tipo de frequência. Pode ser por mês (months) ou por dia (days). Juntamente com a frequência, eles definem o ciclo de parcelas que uma assinatura terá. <br><br>Por exemplo, se a cada quinze dias fosse necessário gerar uma parcela para ser cobrada, seria assim: `auto_recurring.frequency`: 15 y  `auto_recurring.frequency_type`: days |
`auto_recurring.transaction_amount` | Se indicarmos o valor, ele é fixo. Se não preenchermos este campo, entende-se que se trata de uma quantia variável. É permitido um máximo de duas casas decimais separadas por um ponto.|
`auto_recurring.currency_id` (obrigatório) | Identifica a moeda que corresponde ao país. |
`auto_recurring.repetitions` | Indica se a assinatura terá um limite. Se não for especificado, não há limite. Este limite se relaciona com `auto_recurring.frequency` y `auto_recurring.frequency_type`. |
`auto_recurring.free_trial.frequency` | Define um período de teste inicial e retarda a primeira cobrança. Indica o tempo pelo qual o serviço não será cobrado com base no tipo de frequência. Deve ser consistente com `auto_recurring.frequency`. |

> `auto_recurring.free_trial.frequency_type` | Indica o número de parcelas que não serão cobradas pelo serviço. Deve ser consistente com `auto_recurring.frequency_type` |

Reviewer: ```suggestion
`auto_recurring.free_trial.frequency_type` | Indica o número de parcelas que não serão cobradas pelo serviço. Deve ser consistente com `auto_recurring.frequency_type`. |
```

        "transaction_amount": 1100,
        "currency_id": "[FAKER][CURRENCY][ACRONYM]",
        "repetitions": 12,
        "free_trial": {
            "frequency": 1,
            "frequency_type": "months"
        }
    }
}

```
Pronto! Agora podemos criar a assinatura e associá-la ao seu plano.

### Criar assinatura

Uma vez que você tenha gerado seu plano e obtido seu `preapproval_plan_id`, crie a assinatura por API do pagador da seguinte forma: 

[[[
```curl curl --location --request POST 'https://api.mercadopago.com/preapproval \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer ENV_ACCESS_TOKEN' \
--data-raw '{
   "preapproval_plan_id":"2c938084726e18d60172720000000000",
   "card_token_id":"9fca87c7338585abd000111222333444",
   "payer_email":"test_user_XXXX@testuser.com"
   
}'
```
]]]

#### Atributos

Atributo | Definição
--- | ---
`preapproval_plan_id` (obrigatório) | Refere-se ao plano gerado anteriormente.|
`card_token_id` (obrigatório) | Las informações no cartão serão convertidas em um token para enviar os dados com segurança. |
`payer_email` (obrigatório) | E-mail do pagador. |

> WARNING
> 
> Importante
> 

> > Dúvidas sobre como criar o token de pagamento? Encontre todas as informações na seção de __<a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/sdks" target="_blank">Capturar dados do cartão</a>__.

Reviewer: ```suggestion
> Dúvidas sobre como criar o token de pagamento? Encontre todas as informações na seção de __<a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/online-payments/checkout-api/receiving-payment-by-card#bookmark_capture_os_dados_de_cart_o" target="_blank">Capturar dados do cartão</a>__.
```

#### Resposta 
`HTTP Status 200 OK`
```json
{
    "id": "2c938084726e18d60172750000000000",
    "preapproval_plan_id": "2c938084726e18d60170001112223334",
    "payer_id": 100200300,
    "payer_email": "test_user_XXXX@testuser.com",
    "back_url": "http://www.mercadopago[FAKER][URL][DOMAIN]/",
    "collector_id": 10101,
    "application_id": 1234567812345678,
    "status": "authorized",
    "reason": "Plan Pase Mensual Gold",
    "external_reference": "125124513",
    "date_created": "2020-06-02T08:37:42.734-04:00",
    "last_modified": "2020-06-02T08:37:42.735-04:00",
    "init_point":  "https://www.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726e18d60172750000000000",
    "sandbox_init_point": "https://sandbox.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726e18d60172750000000000",
    "auto_recurring": {
        "frequency": 1,
        "frequency_type": "months",
        "transaction_amount": 1100,
        "currency_id": "[FAKER][CURRENCY][ACRONYM]",
        "start_date": "2020-07-02T08:37:42.734-04:00",
        "end_date": "2021-07-02T11:59:52.581-04:00"
}
```

> >Você pode obter mais informações sobre os campos na <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/reference/" target="_blank">Referência de API.</a>.

Reviewer: ```suggestion
>Você pode obter mais informações sobre os campos na <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/reference/" target="_blank">Referência de API</a>.
```

  "reason": "Suscripción Pase Mensual Gold - Particular",
  "status": "pending"
}'
```
]]]

#### Resposta
`HTTP Status 200 OK`
```json

{
    "id": "2c938084726fca480172750000000000",
    "payer_id": 400500600,
    "payer_email": "test_user_XXXX@testuser.com",
    "back_url": "http://www.mercadopago[FAKER][URL][DOMAIN]/",
    "collector_id": 100200300,
    "application_id": 1234567812345678,
    "status": "pending",
    "reason": "Suscripción Pase Mensual Gold - Particular",
    "external_reference": "23546246234",
    "date_created": "2020-06-02T09:07:14.260-04:00",
    "last_modified": "2020-06-02T09:07:14.263-04:00",
    "init_point": "https://www.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726fca480172750000000000",
    "sandbox_init_point": "https://sandbox.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726fca480172750000000000",
    "auto_recurring": {
        "frequency": 1,
        "frequency_type": "months",
        "transaction_amount": 1100,
        "currency_id": "[FAKER][CURRENCY][ACRONYM]",
        "start_date": "2020-06-02T09:07:14.260-04:00",
        "end_date": "2022-07-20T11:59:52.581-04:00"
    },
    "version": 0
}
```

> > Você pode obter mais informações sobre os campos na <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/reference/" target="_blank">Referência de API.</a>.

Reviewer: ```suggestion
> Você pode obter mais informações sobre os campos na <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/reference/" target="_blank">Referência de API</a>.
```

---
indexable: false
---

# Integrar assinaturas   

## Tipos de integração

Há duas formas de integrar assinaturas: 

* __Com um plano associado__: Use esta forma quando você precisar usar a mesma assinatura em ocasiões diferentes e quiser organizá-las em grupos identificáveis. Por exemplo, para uma assinatura mensal e anual de uma academia.

* __Sem um plano associado__: Use esta forma quando você souber que diferentes assinaturas não compartilharão nenhuma característica porque serão muito específicas ou especializadas para cada pagador. Por exemplo, para uma assinatura de um único mês com um desconto específico.

> NOTE
> 
> Conceitos-chave
> 

> > Dúvidas sobre o que é um plano ou outro conceito? Tenha em mãos os <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/guides/online-payments/subscriptions/introduction" target="_blank">conceitos-chave</a> para revisá-los quando necessário.

Reviewer 2: ```suggestion
> Dúvidas sobre o que é um plano ou outro conceito? Tenha em mãos os <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/online-payments/subscriptions/introduction" target="_blank">conceitos-chave</a> para revisá-los quando necessário.
```

        "transaction_amount": 1100,
        "currency_id": "[FAKER][CURRENCY][ACRONYM]",
        "repetitions": 12,
        "free_trial": {
            "frequency": 1,
            "frequency_type": "months"
        }
    }
}

```
Pronto! Agora podemos criar a assinatura e associá-la ao seu plano.

### Criar assinatura

Uma vez que você tenha gerado seu plano e obtido seu `preapproval_plan_id`, crie a assinatura por API do pagador da seguinte forma: 

[[[
```curl curl --location --request POST 'https://api.mercadopago.com/preapproval \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer ENV_ACCESS_TOKEN' \
--data-raw '{
   "preapproval_plan_id":"2c938084726e18d60172720000000000",
   "card_token_id":"9fca87c7338585abd000111222333444",
   "payer_email":"test_user_XXXX@testuser.com"
   
}'
```
]]]

Atributos

Atributo |	Definição
------------ 	|	--------
`preapproval_plan_id (obrigatório)` | Refere-se ao plano gerado anteriormente.|
`card_token_id (obrigatório)` | LAs informações no cartão serão convertidas em um token para enviar os dados com segurança. |
`payer_email (obrigatório)` | E-mail do pagador. |

> WARNING
> 
> Importante
> 

> > Dúvidas sobre como criar o token de pagamento? Encontre todas as informações na seção de __<a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/guides/sdks/official/js/" target="_blank">Capturar dados do cartão.</a>__.

Reviewer 2: ```suggestion
> Dúvidas sobre como criar o token de pagamento? Encontre todas as informações na seção de __<a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/sdks" target="_blank">Capturar dados do cartão.</a>__.
```

#### Resposta 
`HTTP Status 200 OK`
```json
{
    "id": "2c938084726e18d60172750000000000",
    "preapproval_plan_id": "2c938084726e18d60170001112223334",
    "payer_id": 100200300,
    "payer_email": "test_user_XXXX@testuser.com",
    "back_url": "http://www.mercadopago[FAKER][URL][DOMAIN]/",
    "collector_id": 10101,
    "application_id": 1234567812345678,
    "status": "authorized",
    "reason": "Plan Pase Mensual Gold",
    "external_reference": "125124513",
    "date_created": "2020-06-02T08:37:42.734-04:00",
    "last_modified": "2020-06-02T08:37:42.735-04:00",
    "init_point":  "https://www.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726e18d60172750000000000",
    "sandbox_init_point": "https://sandbox.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726e18d60172750000000000",
    "auto_recurring": {
        "frequency": 1,
        "frequency_type": "months",
        "transaction_amount": 1100,
        "currency_id": "[FAKER][CURRENCY][ACRONYM]",
        "start_date": "2020-07-02T08:37:42.734-04:00",
        "end_date": "2021-07-02T11:59:52.581-04:00"
}
```

> >Você pode obter mais informações sobre os campos na <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/reference/" target="_blank">Referência de API.</a>.

Reviewer 2: ```suggestion
>Você pode obter mais informações sobre os campos na <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/reference/" target="_blank">Referência de API.</a>.
```

  "reason": "Suscripción Pase Mensual Gold - Particular",
  "status": "pending"
}'
```
]]]

#### Resposta
`HTTP Status 200 OK`
```json

{
    "id": "2c938084726fca480172750000000000",
    "payer_id": 400500600,
    "payer_email": "test_user_XXXX@testuser.com",
    "back_url": "http://www.mercadopago[FAKER][URL][DOMAIN]/",
    "collector_id": 100200300,
    "application_id": 1234567812345678,
    "status": "pending",
    "reason": "Suscripción Pase Mensual Gold - Particular",
    "external_reference": "23546246234",
    "date_created": "2020-06-02T09:07:14.260-04:00",
    "last_modified": "2020-06-02T09:07:14.263-04:00",
    "init_point": "https://www.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726fca480172750000000000",
    "sandbox_init_point": "https://sandbox.mercadopago.com/mlm/debits/new?preapproval_id=2c938084726fca480172750000000000",
    "auto_recurring": {
        "frequency": 1,
        "frequency_type": "months",
        "transaction_amount": 1100,
        "currency_id": "[FAKER][CURRENCY][ACRONYM]",
        "start_date": "2020-06-02T09:07:14.260-04:00",
        "end_date": "2022-07-20T11:59:52.581-04:00"
    },
    "version": 0
}
```

> > Você pode obter mais informações sobre os campos na <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/reference/" target="_blank">Referência de API.</a>.

Reviewer 2: ```suggestion
> Você pode obter mais informações sobre os campos na <a href="https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/reference/" target="_blank">Referência de API.</a>.
```

`auto_recurring.transaction_amount (obrigatório)` | Valor aplicado à assinatura. |
`auto_recurring.currency_id (obrigatório)` | Identifica a moeda que corresponde ao país. |
`auto_recurring.end_date` | Indica se a assinatura terá um limite. Se não especificado, não há limite. |
`auto_recurring.free_trial.frequency` | Indica o tempo pelo qual o serviço não será cobrado. Deve ser consistente com  `auto_recurring.frequency` |
`auto_recurring.free_trial.frequency_type` | Indica o número de parcelas que não serão cobradas pelo serviço. Deve ser consistente com `auto_recurring.frequency_type` |
`collector_id (obrigatório)` | Identificador do vendedor. |
`payer_email (obrigatório)` | E-mail do pagador.  |
`card_token_id (obrigatório)` | Se a assinatura já foi autorizada, as informações do cartão serão convertidas em um token para enviar os dados com segurança. |

------------
### Próximos passos

> LEFT_BUTTON_RECOMMENDED_PT
>
> Testes
>
> Verifique se suas assinaturas estão configuradas corretamente com os usuários de teste. 
>

> > [Pruebas](http://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/guides/online-payments/subscriptions/testing/)

Reviewer 2: ```suggestion
> [Pruebas](http://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/online-payments/subscriptions/testing/)
```

> RIGHT_BUTTON_RECOMMENDED_PT
>
> Integração avançada
>
> Atualize, altere ou cancele suas assinaturas.
>

> > [Pruebas](http://www.mercadopago[FAKER][URL][DOMAIN]/developers/es/guides/online-payments/subscriptions/advanced-integration/)

Reviewer 2: ```suggestion
> [Pruebas](http://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/guides/online-payments/subscriptions/advanced-integration/)
```
